modules/http-api/src/main/java/org/mule/runtime/http/api/sse/SseRequestContext.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2023 Salesforce, Inc. All rights reserved.
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.http.api.sse;
+
+import org.mule.runtime.http.api.domain.message.request.HttpRequest;
+import org.mule.runtime.http.api.domain.message.response.HttpResponse;
+
+import java.util.concurrent.CompletableFuture;
+
+public interface SseRequestContext {
+
+  HttpRequest getRequest();
```

Comment:
there are some situations in which it could be handy to also have this as a property of the SseClient. Having mixed feelings about it though. let's not do that just yet but marinate on it and tell me what you think

Reply:
The implementation can easily have a reference to the initiator request. If you have a use-case, please tell me and we can add a getter

Reply:
let's leave it as is for now.